### PR TITLE
feat: virtualized-masonry

### PR DESCRIPTION
**Pull request by gkartalis**
### Description

This PR resolves []

#### Acceptance Criteria

-

#### Setup Instructions

-

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by dimatretyak** [comment]
JFYI https://twitter.com/naqvitalha/status/1564867813697024000

**Comment by gkartalis** [comment]
> JFYI https://twitter.com/naqvitalha/status/1564867813697024000

ooooh exciting!!! 

alt

File: src/app/Components/VirtualizedMasonry.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          height={row.height}
+          flexDirection="row"
+          mx={gutter / -2}
+          mt={index < numColumns ? gutter / -2 : 0}
+          mb={index === rowData.length - 1 ? gutter / -2 : 0}
+        >
+          {columns.map((_val, i) => (
+            <Flex key={`${index}-${i}`} pointerEvents="box-none" flex={1} p={gutter / 2}>
+              {i === row.columnIndex && renderItem ? (
+                <Flex width={brickWidth} height={row.brick.height}>
```
Comment (pvinis) [inline src/app/Components/VirtualizedMasonry.tsx:89]: this was a really important part! we should fork https://github.com/iftechio/react-native-virtualized-masonry and fix this.

File: src/app/Components/VirtualizedMasonry.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+type Brick<ItemT> = {
+  item: ItemT
+  height: number
+}
+
+type Row<ItemT> = {
+  brick: Brick<ItemT>
+  columnIndex: number
+  height: number
```
Comment (pvinis) [inline src/app/Components/VirtualizedMasonry.tsx:19]: also clearing up the types was huge. there are two `height`s! one for the brick, one for the row, and using the wrong height messes things up.

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx [hunk 3]
```diff
@@ -90,7 +87,6 @@ export const Artwork: React.FC<ArtworkProps> = ({
   partnerNameTextStyle,
   updateRecentSearchesOnTap = false,
 }) => {
-  const itemRef = useRef<any>()
```
Comment (pvinis) [inline src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:93]: unused ref 🤔

**Comment by pvinis** [comment]
ok this works great here! feel free to continue with this PR, or we can finish it up together, but at least masonry works now. thank you for working on this!

File: src/app/Scenes/Search/SearchArtworksGrid.tsx [hunk 4]
```diff
@@ -160,12 +155,54 @@ export const SearchArtworksGridPaginationContainer = createPaginationContainer(
...
           }
+          pageInfo {
+            hasNextPage
+            startCursor
+            endCursor
+          }
           edges {
             node {
               id
+              title
```
Comment (olerichter00) [inline src/app/Scenes/Search/SearchArtworksGrid.tsx:166]: **question:** Shouldn't all these fields go into `ArtworkGrid`'s fragment?

Comment (olerichter00) [reply src/app/Scenes/Search/SearchArtworksGrid.tsx:166]: 🚀

**Review by gkartalis — COMMENTED** [review]
Did a pairing session with @pvinis today on this, we had a weird flashing - rerendering issue when fetching more data with the new MasonryList. 

- When replacing the Masonry with a Flatlist the rerender didn't happen 🤔 
- tried to render only a text with title instead of artwork component, flashing didn't happen
- replaced artwork component return with an early return in artwork (text/image) didn't flash

Maybe we need to try `@welldone-software/why-did-you-render` in order to understand the reason why.

**Comment by gkartalis** [comment]
Closing this since we decided to try out the `@shopify/flash-list` one

**Pull request closed without merging**